backport: configfs: const struct config_item_type

diff --git a/most/configfs.c b/most/configfs.c
--- a/most/configfs.c
+++ b/most/configfs.c
@@ -383,7 +383,7 @@ static struct configfs_item_operations mdev_link_item_ops = {
 	.release		= mdev_link_release,
 };
 
-static const struct config_item_type mdev_link_type = {
+static struct config_item_type mdev_link_type = {
 	.ct_item_ops	= &mdev_link_item_ops,
 	.ct_attrs	= mdev_link_attrs,
 	.ct_owner	= THIS_MODULE,
@@ -433,7 +433,7 @@ static struct configfs_group_operations most_common_group_ops = {
 	.make_item	= most_common_make_item,
 };
 
-static const struct config_item_type most_common_type = {
+static struct config_item_type most_common_type = {
 	.ct_item_ops	= &most_common_item_ops,
 	.ct_group_ops	= &most_common_group_ops,
 	.ct_owner	= THIS_MODULE,
@@ -535,7 +535,7 @@ static struct configfs_group_operations most_snd_grp_group_ops = {
 	.make_item	= most_snd_grp_make_item,
 };
 
-static const struct config_item_type most_snd_grp_type = {
+static struct config_item_type most_snd_grp_type = {
 	.ct_item_ops	= &most_snd_grp_item_ops,
 	.ct_group_ops	= &most_snd_grp_group_ops,
 	.ct_attrs	= most_snd_grp_attrs,
@@ -573,7 +573,7 @@ static struct configfs_group_operations most_sound_group_ops = {
 	.make_group	= most_sound_make_group,
 };
 
-static const struct config_item_type most_sound_type = {
+static struct config_item_type most_sound_type = {
 	.ct_group_ops	= &most_sound_group_ops,
 	.ct_owner	= THIS_MODULE,
 };
