backport: networking: ether_addr_equal

diff --git a/most/net/net.c b/most/net/net.c
--- a/most/net/net.c
+++ b/most/net/net.c
@@ -558,7 +558,7 @@ static void on_netinfo(struct most_interface *iface,
 				    m[0], m[1], m[2], m[3], m[4], m[5]);
 			ether_addr_copy(dev->dev_addr, m);
 			netif_dormant_off(dev);
-		} else if (!ether_addr_equal(dev->dev_addr, m)) {
+		} else if (memcmp(dev->dev_addr, m, ETH_ALEN)) {
 			netdev_warn(dev, "reject mac %02x-%02x-%02x-%02x-%02x-%02x\n",
 				    m[0], m[1], m[2], m[3], m[4], m[5]);
 		}
