backport: networking: alloc_netdev

diff --git a/most/net/net.c b/most/net/net.c
--- a/most/net/net.c
+++ b/most/net/net.c
@@ -312,7 +312,7 @@ static int comp_probe_channel(struct most_interface *iface, int channel_idx,
 	nd = get_net_dev(iface);
 	if (!nd) {
 		dev = alloc_netdev(sizeof(struct net_dev_context), "meth%d",
-				   NET_NAME_UNKNOWN, most_nd_setup);
+				   most_nd_setup);
 		if (!dev) {
 			ret = -ENOMEM;
 			goto unlock;
